src/modules/users/components/AvatarDropdown/AvatarDropdown.tsx

```diff
@@ -15,20 +15,40 @@ import styles, {
   horizontalOffset,
   verticalOffset,
 } from './AvatarDropdown.css';
+import AvatarDropdownPopoverMobile from './AvatarDropdownPopoverMobile';
+import { SimpleMessageValues } from '~types/index';
+import { UserTokenData } from '~types/tokens';
 
 const UserAvatar = HookedUserAvatar();
 
+export interface AppState {
+  previousWalletConnected: string | null;
+  attemptingAutoLogin: boolean;
+  userDataLoading: boolean;
+  userCanNavigate: boolean;
+}
+
 interface Props {
   onlyLogout?: boolean;
   colony: Colony;
+  userData: Partial<LoggedInUser>;
+  spinnerMsg: SimpleMessageValues;
+  userTokenData: UserTokenData;
+  appState: AppState;
 }
 
 const displayName = 'users.AvatarDropdown';
 
-const AvatarDropdown = ({ onlyLogout = false, colony }: Props) => {
-  const { username, walletAddress, ethereal } = useLoggedInUser();
```

Comment:
Instead of calling useLoggedInUser again (already called in parent), I pass down relevant variables in an object.